src/store/actions.ts

```diff
@@ -883,3 +639,433 @@ export const setStaffPayment: ActionIF = ({ commit }, staffPayment: StaffPayment
+ */
+export const getMatchesExact = async (
+  { commit },
+  token: string,
+  cleanedName: string
+): Promise<Array<ConflictListItemI>> => {
+  const exactResp = await axios.get('/exact-match?query=' + cleanedName, {
+    headers: { Authorization: `Bearer ${token}`, 'Content-Type': 'application/json' }
+  }).catch(() => {
+    commit('mutateNameCheckErrorAdd', NameCheckErrorType.errorExact)
+    return null
+  })
+  return exactResp?.data ? parseExactNames(exactResp.data) : []
+}
+export const getMatchesSimilar = async (
+  { commit },
+  token: string,
+  cleanedName: string,
+  exactNames: Array<ConflictListItemI>
+): Promise<Array<ConflictListItemI>> => {
+  const synonymResp = await axios.get('/requests/synonymbucket/' + cleanedName + '/*', {
+    headers: { Authorization: `Bearer ${token}`, 'Content-Type': 'application/json' }
+  }).catch(() => {
+    commit('mutateNameCheckErrorAdd', NameCheckErrorType.errorSimilar)
+    return null
+  })
+  if (synonymResp?.data) synonymResp.data.exactNames = exactNames || []
+  return synonymResp?.data ? parseSynonymNames(synonymResp.data) : []
+}
+export const getMatchesRestricted = async (
+  { commit },
+  token: string,
+  cleanedName: string
+): Promise<ParsedRestrictedResponseIF> => {
+  const restrictedResp = await axios.get(`/documents:restricted_words?content=${cleanedName}`, {
+    headers: { Authorization: `Bearer ${token}`, 'Content-Type': 'application/json' }
+  }).catch(() => {
+    commit('mutateNameCheckErrorAdd', NameCheckErrorType.errorRestricted)
+    return null
+  })
+  return restrictedResp?.data
+    ? parseRestrictedWords(restrictedResp.data)
+    : { restrictedWords: [], conditionalWords: [] }
+}
+
+export const getNameAnalysis: ActionIF = async (
+  { commit, getters },
+  options: { xpro: boolean, designationOnly: boolean }
+) => {
+  const name = getters.getFullName
+  try {
+    if (options.designationOnly) commit('mutateAnalyzeDesignationPending', true)
+    else commit('mutateAnalyzeStructurePending', true)
+    commit('resetRequestExaminationOrProvideConsent')
+    const params: NewRequestNameSearchI = {
+      name: name,
+      location: getters.getLocation,
+      entity_type_cd: getters.getEntityTypeCd,
+      request_action_cd: getters.getRequestActionCd,
+      jurisdiction: options.xpro ? 'XPRO' : 'BC',
+      analysis_type: options.designationOnly ? 'designation' : 'structure'
+    }
+    const analysis = await NamexServices.nameAnalysis(params)
+    // verify the user did not start a new search on a different name
+    if (name === getters.getFullName) {
+      const json = analysis
+      commit('mutateAnalysisJSON', json)
+      if (Array.isArray(json.issues) && json.issues.length > 0) {
+        for (let i = 0; i < json.issues.length; i++) {
+          switch (json.issues[i].issue_type) {
+            case 'add_descriptive':
+              commit('mutateMissingDescriptive', true)
+              continue
```

Comment:
On the last "for" item, what does `continue` do? Just drop out of the for loop?

Reply:
yes it was unnecessary

Reply:
Oh no, that's not what I meant. (I was wondering what happens after the last iteration -- does the code execute after the `continue`? But I think it actually goes to the code after the for block.) So my original comment is not an issue.

Removing the `continue` in the default block is OK since that's the end of the for block anyway, so it will "continue" with any other for iterations, as desired.